awx/main/tests/unit/test_indirect_query_discovery.py

```
@@ -0,0 +1,436 @@
+
+    @mock.patch('awx.playbooks.library.indirect_instance_count.files')
+    def test_exact_match_preferred(self, mock_files):
+        """AC7.4: Exact version match is preferred over fallback version."""
+        from awx.playbooks.library.indirect_instance_count import find_external_query_with_fallback
+
+        mock_exact_file = mock.Mock()
+        mock_exact_file.exists.return_value = True
+        mock_exact_file.open.return_value.__enter__ = mock.Mock(return_value=StringIO('exact_version_query'))
+        mock_exact_file.open.return_value.__exit__ = mock.Mock(return_value=False)
+
+        queries_dir = create_queries_dir_mock(lambda f: mock_exact_file)
```

Ruff (0.15.2)

[warning] Unused lambda argument: `f`

(ARG005)